pkg/tfbridge/provider.go

```diff
@@ -949,12 +952,35 @@ func (p *Provider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*pulum
 
 	ic := newIgnoreChanges(ctx, schema, fields, olds, news, req.GetIgnoreChanges())
 
+	// if we have type errors that were generated during check
+	// we don't want to log the panic. In the future these type errors
+	// will hard fail during check and we will never make it to create
+	//
+	// We do our best to restrict catching panics to where the panic will
+	// occur. The type checking will catch panics that occur as part
+	// of the `Diff` function
+	recoverFunc := func(r interface{}) {
+		if _, ok := p.hasTypeErrors[urn]; !ok {
+			panic(r)
+		}
+	}
+	defer func() {
+		if r := recover(); r != nil {
+			recoverFunc(r)
+		}
+	}()
+
 	diff, err := p.tf.Diff(ctx, res.TFName, state, config, shim.DiffOptions{
```

Comment:
contrary to what you'd assume given the name, this interface actually calls into our code - we have 3-4 implementations of Diff under `tfshim/`.

Not sure if we care about catching possible panics there - I don't really know if all of our code would handle mistyped input so, it's probably fine here.